src/logic/calculate.js

```diff
@@ -0,0 +1,42 @@
+import operate from './operate';
+
+const calculate = (object, buttonName) => {
+  let { total, next, operation } = object;
+
+  const parsingInt = string => parseInt(string, 10);
+
+  switch (buttonName) {
+    case '+/-':
+      operation = '÷';
+      total = operate(parsingInt(total), -1, operation);
+      next = operate(parsingInt(next), -1, operation);
+      break;
+    case 'AC':
+      total = null;
+      next = null;
+      operation = null;
+      break;
+    case '=':
+    case 'x':
+    case '÷':
+    case '+':
+    case '-':
+      operation = buttonName;
+      total = operate(parsingInt(total), parsingInt(next), operation);
+      break;
+    case '%':
+      operation = buttonName;
+      total = operate(parsingInt(total), 100, operation);
+      break;
```

Comment:
Suppose I enter '%', based on the current condition on the operation I will get 'error' as a result. It should not interact like that. The operation should be division operation, isn't it?

Reply:
Somehow I changed at some point, thank you, I'll look for it.